src/utiltime.cpp

```diff
@@ -50,6 +50,12 @@ int64_t GetTimeMillis()
 
 int64_t GetTimeMicros()
 {
+    int64_t mocktime = nMockTime.load(std::memory_order_relaxed);
```

Comment:
Wonder if we could do the same for `GetTimeMillis` which actually use `nMockTime` directly rather than goes through `nMockTime.load`

Reply:
we could.  The cast operator is the same as .load() (https://en.cppreference.com/w/cpp/atomic/atomic/operator_T).  However, load() was used here so that the std::memory_order_relaxed flag could be used which may marginally increase performance, since it guarantees nothing but atomic access.  You are correct that GetTimeMillis would also be marginally faster, and since this PR is meant to make the time logic "perfect" I should use .load there as well.

Reply:
committed.